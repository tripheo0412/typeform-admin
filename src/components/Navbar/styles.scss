@import "../../assets/styles/variables";
@import "../../assets/styles/mixins";
@import "../../assets/styles/normalize";

.navbar {
  @include font-size-m;
  font-family: $font-family;
  font-weight: 300;
  display: flex;
  border-bottom: 1px solid $secondary;
  background-color: $white;
  justify-content: space-between;
  align-items: center;
  height: 80px;

  &__left {
    flex: 1;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    &--desktop {
      display: flex;
      justify-content: center;
      align-items: center;
    }
    &--logo {
      margin: $spacing-01;
      cursor: pointer;
    }

    &--title {
      padding-right: $spacing-00-25;
      text-decoration: none;
      text-transform: capitalize;
      color: $grey-light;
    }
    &--mobile {
      display: none;
    }
  }
  .template--name {
    color: $primary;

    &::before {
      content: "\002F";
      font-size: 15px;
      padding-right: 2px;
      color: $grey-light;
    }
  }
  &__navlinks {
    display: flex;
    justify-content: center;
    list-style-type: none;

    li {
      margin: $spacing-01;
    }
    a {
      text-decoration: none;
      text-transform: capitalize;
      color: $grey-light;

      &:hover {
        color: $primary;
      }
      &:active {
        color: $danger;
      }
    }
  }
  &__center {
    display: flex;
    flex: 1;
    justify-content: flex-end;
    align-items: center;
    margin-right: $spacing-01;
    &__mobile {
      display: none;
    }

    button {
      margin-right: $spacing-01;
    }
  }
  &__workspace {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-left: $spacing-01;
    padding-right: $spacing-01;
  }
}

@include for-tablet-portrait-down {
  .dot__menu {
    margin-right: $spacing-01;
  }
  .navbar__left--desktop {
    display: none;
  }

  .navbar__left--mobile {
    display: block;

    button {
      background: transparent;
      margin-right: $spacing-00-25;
      margin-left: $spacing-00-25;
      border: none;
      &:after {
        content: "\003C";
        font-size: 20px;
        color: $grey-light;
      }
    }
  }
  .navbar__navlinks {
    display: none;
  }
  .navbar__center {
    margin: 0;
    .button {
      padding: $spacing-00-50 0;
      min-width: 40%;
    }
    .avatar {
      display: none;
    }
  }
}

@include for-desktop-up-1 {
  .dot__menu {
    display: none;
  }
  .navbar__navlinks--mobile {
    display: none;
  }
}
