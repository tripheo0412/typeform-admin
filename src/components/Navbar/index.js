// @flow
import React from 'react';
import IconDotMenu from '../IconDotMenu';
import Button from '../Button';
import Avatar from '../Avatar';
import { DropDown } from '../DropDownMenu';
import './styles.scss';

type Props = {
  isWorkspase?: boolean,
  templateName?: string,
  handleSave?: () => void,
  handleView?: () => void,
  handleBack?: () => void,
};

const links = [
  {
    title: 'edit',
    handleClick: () => alert('link'),
    linkUrl: '#',
  },
  {
    title: 'share',
    handleClick: () => alert('link'),
    linkUrl: '#',
  },
  {
    title: 'results',
    handleClick: () => alert('link'),
    linkUrl: '#',
  },
];
export const Navbar = ({
  templateName,
  handleSave,
  handleView,
  handleBack,
  isWorkspase,
}: Props) => {
  const navLinksTarget = links.map(link => (
    // TODO: change the a link to be <Link to="l.linkUrl"/>
    <li key={link.title}>
      <a href={link.linkUrl} target="">
        {link.title}
      </a>
    </li>
  ));

  const workspace = () => (
    <div className="navbar navbar__workspace">
      <div className="navbar__left--logo">
        <placeholder>LOGO</placeholder>
      </div>

      <Avatar
        withMenu
        initialName="vj"
        imgUrl="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50"
      />
    </div>
  );

  return (
    (isWorkspase && workspace()) || (
      <div className="navbar">
        <div className="navbar__left">
          <div className="navbar__left--desktop">
            <div className="navbar__left--logo">
              <placeholder>LOGO</placeholder>
            </div>
            <a className="navbar__left--title" href="/" target="">
              my workspace
            </a>
            <span className="template--name">{templateName}</span>
          </div>
          <div className="navbar__left--mobile">
            <button onClick={handleBack} type="submit"></button>
            {templateName}
          </div>
        </div>
        <ul className="navbar__navlinks">{navLinksTarget}</ul>
        <div className="navbar__center">
          <Button
            variant="primary"
            size="sm"
            label="Save"
            onClick={handleSave}
          />
          <Button
            variant="secondary"
            size="sm"
            label="View"
            onClick={handleView}
          />
          <Avatar
            withMenu
            initialName="vj"
            imgUrl="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50"
          />
        </div>
        <div className="dot__menu">
          <IconDotMenu direct="left">
            <DropDown options={links} />
          </IconDotMenu>
        </div>
      </div>
    )
  );
};

Navbar.defaultProps = {
  handleSave: () => {},
  handleView: () => {},
  handleBack: () => {},
};

export default Navbar;
