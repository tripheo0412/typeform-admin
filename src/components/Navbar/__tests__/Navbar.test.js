import React from 'react';
import { shallow } from 'enzyme';

import Navbar from '..';

describe('navbar component', () => {
  let component;
  it('debug', () => {
    const logo = {
      src: '',
      alt: 'my logo',
    };

    const avatar = {
      src: '',
      alt: 'my logo',
    };

    const settings = [
      {
        title: 'Settings',
        handleClick: () => alert('Settings'),
      },
      {
        title: 'Account',
        handleClick: () => alert('Accout'),
      },
      {
        title: 'Your Profile',
        handleClick: () => alert('Profile'),
      },
    ];

    component = shallow(
      <Navbar logo={logo} avatar={avatar} settings={settings} />
    );
    expect(component).toMatchSnapshot();
  });
});
