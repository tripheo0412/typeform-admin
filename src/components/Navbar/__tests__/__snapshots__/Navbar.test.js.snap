// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component debug 1`] = `ShallowWrapper {}`;
